include/fmt/core.h

```diff
@@ -1468,13 +1548,38 @@ class dynamic_format_arg_store
   friend class basic_format_args<Context>;
 
   unsigned long long get_types() const {
-    return internal::is_unpacked_bit | data_.size();
+    return internal::is_unpacked_bit | data_.size() |
+           (named_info_.empty() ? 0ULL
+                                : static_cast<unsigned long long>(
+                                      internal::has_named_args_bit));
+  }
+
+  const basic_format_arg<Context>* data() const {
+    return named_info_.empty() ? data_.data() : data_.data() + 1;
   }
 
   template <typename T> void emplace_arg(const T& arg) {
     data_.emplace_back(internal::make_arg<Context>(arg));
   }
 
+  template <typename T>
+  void emplace_arg(const internal::named_arg<T, char_type>& arg) {
+    if (named_info_.empty()) {
+      data_.insert(
+          data_.begin(),
+          {static_cast<const internal::named_arg_info<char_type>*>(nullptr),
+           0});
+    }
+    data_.emplace_back(internal::make_arg<Context>(
+        static_cast<const internal::unwrap_reference_t<T>&>(arg.value)));
```

Comment:
What is this cast for?

Reply:
This is to support `named_arg<std::reference_wrapper<T>>` reference_wrapper convertable to T&, while for non-wrapped it's a cast to itself ref.

Reply:
I suggest replacing this with an `unwrap` function that unwraps `reference_wrapper` or returns the reference to the argument if it's not one. This will make the code slightly clearer.